frontend/src/js/asyncHandlers.js

```diff
@@ -0,0 +1,209 @@
+
+    /* 
+        ---- GET DATA SCRIPTS ----
+    */    
+
+    var rawGeo, rawLar, rawMinority, rawData, 
+    isUIBlocked = false,
+    dataStore = {};
+    dataStore.tracts = {};
+    
+    // Get the census tracts that are in bounds for the current map view. Return a promise.
+    function getTractsInBounds(bounds, geoType){
+
+        $('#bubbles_loading').show();
+
+        // Create the appropriate URL path to return values
+        var endpoint = '/api/tractCentroids/', 
+            params = {};
+
+        if( bounds && typeof bounds === 'object' ){
+            params.neLat = bounds.neLat;
+            params.neLon = bounds.neLon;
+            params.swLat = bounds.swLat;
+            params.swLon = bounds.swLon;
+        }
+
+        if( geoType ){
+            params.geoType = geoType;
+        }
+
+        if ( urlParam('metro') ){
+            params.metro = urlParam('metro');
+        } else {
+            console.log("No metro area provided");
+        }
+
+        // Set the lender parameter based on the current URL param
+        if ( urlParam('lender') ){
+            params['lender'] = urlParam('lender');
+        } else {
+            console.log(' Lender parameter is required.');
+            return false;
+        }
+
+        return $.ajax({
+            url: endpoint, data: params, traditional: true,
+            success: console.log('tract Get successful')
+        }).fail( function( status ){
+            console.log( 'no data was available at' + endpoint + '. status: ' + status );
+        });
+
+    }    
+
+    // Get minority and LAR data for census Tracts within the bounding box, for a specific criteria (actionTaken)
+    // Return a promise.
+    function getTractData( actionTakenVal, bounds, geoType ){
+        $('#bubbles_loading').show();
+        var endpoint = '/api/all/',
+            params = { year: 2013,
+                        'lh': false,
+                        'peers': false,
```

Comment:
No need to pass in 'false' value for params peers and lh. Endpoint checks to see if they are = 'true' and will result to false if nothing is passed in.